rsconf/package_data/rsconf/rsconf.sh

```diff
@@ -309,6 +312,24 @@ rsconf_install_symlink() {
     rsconf_service_file_changed "$new"
 }
 
+rsconf_is_perl_rpm_rollback() {
+    declare base_rpm=$1
+    declare previous_rpm=$2
+    declare this_rpm=$3
+    declare versions=()
+    for rpm in $previous_rpm $this_rpm; do
+        if [[ $rpm =~ ^$base_rpm\-([[:digit:]]{8}\.[[:digit:]]{6}-[[:digit:]])\. ]]; then
```

Comment:
So, the caller of `rsconf_perl_rpm_version` would have to check the return value and throw it's own `install_err`, wouldn't it? Not sure it's worth pushing the version parsing into a function in that case.

Also, I wasn't throwing an error if the pattern didn't match because if this is the first time the rpm is being installed, `previous_rpm` would be `package $base is not installed`. I suppose it might be better to check for that.

Reply:
The rollback case only happens if it is already installed. That should be checked, because it will be confusing. I think the code should look like:

```bash
rsconf_is_perl_rpm_rollback() {
    declare prev_version=$1
    declare install_version=$2
    if ! rsconf_perl_rpm_version prev_version "$prev_version"; then
        return 1
    fi
    rsconf_perl_rpm_version install_version "$install_version"
    if [[  ! ${install_version:-} ]]; then
        install_error "internal error with rsconf_perl_rpm_version parsing (install_version=$2)"
    fi
    [[ $prev_version > $install_version ]]
}

rsconf_perl_rpm_version() { 
    declare var=$1
    declare rpm=$2
    if ! [[ $rpm =~ -([[:digit:]]{8}\.[[:digit:]]{6}-[[:digit:]])\. ]]; then
        if [[ $rpm =~ not.installed ]]; then
            return 1
        fi
        install_err "invalid version format rpm=$rpm"
    fi
    eval "$var='${BASE_REMATCH[1]}'"
}
```

Reply:
b70d5af